src/pages/VestingClaim.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { ReactElement, useEffect, useState } from "react"
+import { SDL_TOKEN, SDL_TOKEN_ADDRESSES } from "../constants"
+import { Trans, useTranslation } from "react-i18next"
+import { commify, formatBNToString, getContract } from "../utils"
+import { notifyCustomError, notifyHandler } from "../utils/notifyHandler"
+import { BigNumber } from "@ethersproject/bignumber"
+import Button from "../components/Button"
+import INVESTOR_EMPLOYEE_VESTING_CONTRACT_ABI from "../constants/abis/vesting.json"
+import SDL_TOKEN_ABI from "../constants/abis/sdl.json"
+import { Sdl } from "../../types/ethers-contracts/Sdl"
+import TopMenu from "../components/TopMenu"
+import { Vesting } from "../../types/ethers-contracts/Vesting"
+import { Zero } from "@ethersproject/constants"
+import logo from "../assets/icons/logo.svg"
+import plusIcon from "../assets/icons/plus.svg"
+import styles from "../components/TokenClaimModal.module.scss"
+import { useActiveWeb3React } from "../hooks"
+import useAddTokenToMetamask from "../hooks/useAddTokenToMetamask"
+
+function VestingClaim(): ReactElement {
+  const { t } = useTranslation()
+  const { account, chainId, library } = useActiveWeb3React()
+  const { addToken, canAdd } = useAddTokenToMetamask({
+    ...SDL_TOKEN,
+    icon: `${window.location.origin}/logo.svg`,
+  })
+
+  const [isValidBeneficiary, setIsValidBeneficiary] = useState(false)
+  const [claimableVestedAmount, setClaimableVestedAmount] = useState(Zero)
+  const [remainingAmount, setRemainingAmount] = useState(Zero)
+  const [vestingContract, setVestingContract] = useState<Vesting>()
+
+  const formatAmount = (amount: BigNumber): string =>
+    commify(formatBNToString(amount, 18, 2))
+
+  useEffect(() => {
+    const fetchBeneficiaries = async () => {
+      if (!library || !chainId || !account) return
+      const SDLContract = getContract(
+        SDL_TOKEN_ADDRESSES[chainId],
+        SDL_TOKEN_ABI,
+        library,
+        account,
+      ) as Sdl
+      const vestingContractDeployedFilter = SDLContract.filters.VestingContractDeployed(
+        null,
+        null,
+      )
+      try {
+        const events = await SDLContract.queryFilter(
+          vestingContractDeployedFilter,
+        )
+        const currentBeneficiaryEvents = events.filter(
+          (event) => event.args?.beneficiary === account,
+        )
+        setIsValidBeneficiary(currentBeneficiaryEvents.length > 0)
+        if (isValidBeneficiary) {
```

Comment:
I think this will cause this effect to be run once more than expected, since `isValidBeneficiary` is both set by this effect and a dependency of it. To fix, remove it as an effect dep and just store / read `currentBeneficiaryEvents.length > 0` as a variable within the effect.

Reply:
nice find, this doesn't need to be an effect dep